packages/actor-query-operation-leftjoin/lib/ActorQueryOperationLeftJoin.ts

```diff
@@ -42,6 +45,15 @@ export class ActorQueryOperationLeftJoin extends ActorQueryOperationTypedMediate
         .transform({
           autoStart: false,
           transform: async(bindings: Bindings, done: () => void, push: (item: Bindings) => void) => {
+            const keysArefulfilled = variables.length > 0 ?
+              variables.every(variable => bindings.has(variable.value)) :
```

Comment:
We need to look only for the variables in the right-hand stream (the one inside the optional block) I think.

Reply:
The checking needed to be done in the left hand stream. That is where bindings were skipped.